CMakeLists.txt

```diff
@@ -337,9 +342,21 @@ if(WITH_PROMETHEUS)
   endif()
 endif()
 
+if(WITH_ABSEIL)
+  find_package(absl CONFIG REQUIRED)
+endif()
+
 if(WITH_OTLP_GRPC OR WITH_OTLP_HTTP)
-  set(protobuf_MODULE_COMPATIBLE ON)
   find_package(Protobuf)
+  if(Protobuf_VERSION AND Protobuf_VERSION VERSION_GREATER_EQUAL "3.22.0")
+    if(NOT WITH_ABSEIL)
+      message(
+        FATAL_ERROR
```

Comment:
Seems with new protobuf, the main SDK will no longer buildable with `WITH_ABSEIL=0` so this will be a required dependency. Can we separate the abseil dependency to protobuf related library only?

Reply:
protobuf has a internal reference to abseil and will use it when building without `-Dprotobuf_ABSL_PROVIDER=package`, just like gRPC.I think it's better to keep the same version for protobuf, gRPC and otel-cpp.